bot/src/main/java/de/chojo/gamejam/data/wrapper/jam/TimeFrame.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *     SPDX-License-Identifier: AGPL-3.0-only
+ *
+ *     Copyright (C) 2022 DevCord Team and Contributor
+ */
+
+package de.chojo.gamejam.data.wrapper.jam;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+public record TimeFrame(ZonedDateTime start, ZonedDateTime end) {
+    public TimeFrame {
+        if (start.isAfter(end)) {
+            throw new IllegalStateException("Start time is after end time.");
+        }
+    }
```

Comment:
This logic should be moved out of the constructor and into a static factory method

Reply:
Validation in constructors are actually fine imo.
This is only about validating but not changing, altering or creating new data

Reply:
I like to refer here to the  secure coding guidelines of for java se (https://www.oracle.com/java/technologies/javase/seccodeguide.html) section 7 Object Construction. But since this class is an record and therefore a final class, this shouldn't be a problem as i see it right now.